### PR TITLE
Updated Introduction > Key Concepts

### PR DESCRIPTION
**Pull request by 4katie**
Signed-off-by: Katie <sa951009@gmail.com>

# Issue #111 
- Updated `Introduction > Key Concepts`
- Expanded slightly on `Introduction > For Business `

### Changes
- Changed the code images with code blocks
- Improved the explanations on `Introduction > Key Concepts` by elaborating on some of the concepts
- Fixed minor typos

### Flags
For the `model` code block, the `cto` language is currently undefined, so I used `ergo` instead. Works perfectly fine, but may want to update this when concerto is supported in the future. 

### Related Issues
- Issue #111 
- PR #207

### CODE REVIEW

File: docs/accordproject-concepts.md [hunk 1]
```diff
@@ -5,31 +5,31 @@ title: Key Concepts
...
 
 - **The Template Text**: the natural language of the template
-- **The Template Model**: the data model that backs the template 
+- **The Template Model**: the data model that backs the template
 - **The Template Logic**: the executable business logic for the template
 
 ![Accord Project Template](assets/020/template.png)
 
-When combined these three elements allow templates to be edited, validated, verified, and then executed on any computer platform (on your own machine, on a Cloud platform, on Blockchain, etc).
+The three components (Text - Model - Logic) can also be intuitively understood as a **linear progression**, from _human-readable_ legal text to _machine-readable_ code. When combined these three elements allow templates to be edited, validated, verified, and then executed on any computer platform (on your own machine, on a Cloud platform, on Blockchain, etc).
```
Comment (jeromesimeon) [inline docs/accordproject-concepts.md]: The mathematician in me is shaking its head. How about `progression` or `incremental progression`?

File: docs/accordproject-concepts.md [hunk 2]
```diff
@@ -113,19 +132,41 @@ _More information about Concerto can be found in the [Concerto Modeling](model-c
...
+      status : status,
+      shipper : contract.shipper,
+      receiver : contract.receiver
+    }
+  }
+}
+```
 
-That logic specify what conditions should be met for a delivery to be accepted (checking whether the delivery is timely or not, has a mandatory inspection for that delivery occurred or not, etc).
+The above logic specify what conditions should be met for a delivery to be accepted. The code above checks whether delivery has already been made; whether the acceptance is timely, within the specified inspection date; and whether the inspection has passed or not.
```
Comment (jeromesimeon) [inline docs/accordproject-concepts.md]: `specify` --> `specifies`

File: docs/accordproject-concepts.md (hunk 1 above)
Comment (4katie) [reply docs/accordproject-concepts.md]: Hahaha sounds good ;) (makes note to self - nothing with _linear_ next time...)

File: docs/accordproject-concepts.md [hunk 3]
```diff
@@ -5,31 +5,31 @@ title: Key Concepts
...
+> Note that we use the computing term 'execution' here, which means run by a computer. This is distinct from the legal term 'execution', which usually refers to the process of signing an agreement.
 
 ### Cicero
 
-The main software implementation for the Accord Project is called [Cicero](https://github.com/accordproject/cicero). It allows users of the accord project to create and manipulate software templates which tie together all three components in the triangle above (i.e., the text, model and logic).
+The main software implementation for the Accord Project is called [Cicero](https://github.com/accordproject/cicero). It allows users of the Accord Project to create and manipulate software templates which tie together all three components in the triangle above (i.e., the text, model and logic).
 
-_More information about how to install Cicero and get started with Accord project templates can be found in the [Installation](accordproject-installation) Section of this documentation._
+_More information about how to install Cicero and get started with Accord Project templates can be found in the [Installation](accordproject-installation) Section of this documentation._
```
Comment (jeromesimeon) [inline docs/accordproject-concepts.md]: I'm pretty sure this link is wrong and should be `started-installation`. Would be worth checking that all the links work.

File: docs/accordproject-developers.md [hunk 4]
```diff
@@ -3,4 +3,22 @@ id: accordproject-developers
 title: For Developers
 ---
 
-**TBD**
+## Why is the Accord Project relevant for developers?
+The Accord Project provides a universal format for smart legal contracts, and this format is embodied in a variety of open source projects that comprise the Accord Project technology stack. Input from developers are crucial for the Accord Project.
+
+Developers can contribute by translating legal text into corresponding computer code, creating Accord Project templates to be used by lawyers and businesses. In addition, developers can provide input on the development of the technology stack components: language, models, templating, and other tools.
```
Comment (jeromesimeon) [inline docs/accordproject-developers.md]: Not sure about the word `translating` (I wished I had something concrete to offer as alternative), but it should probably rewritten a bit.

File: docs/accordproject-developers.md [hunk 5]
```diff
@@ -3,4 +3,22 @@ id: accordproject-developers
...
+
+If this interests you, please visit our [Technology Working Group](https://www.accordproject.org/working-groups/technology) page, and join our [slack channel](https://accord-project-slack-signup.herokuapp.com/)!
+
+
+## How to use this documentation?
+
+If you are new to the Accord Project, please read the [Key Concepts](accordproject-concepts) page. This will allow you to understand the three components of a template (text, model, and logic) and how they work together.
+
+If you want to create a template yourself, please see the [Legal-Tech Tutorials - Authoring in Template Studio](tutorial-latedelivery) page for a step-by-step guide on how to create your first template on template studio. If you want to author, validate, and run Accord Project templates _locally_, please visit our [Install Cicero](https://docs.accordproject.org/docs/next/started-installation.html) page for instructions.
```
Comment (jeromesimeon) [inline docs/accordproject-developers.md]: I would split this part three paragraphs:
1. a paragraph pointing to the getting started guide (same link) to quickly get up and running as a developer
2. the additional information for legal tech and 
3. the `dive into our technology stack` part as is

File: docs/accordproject-laywers.md [hunk 6]
```diff
@@ -3,4 +3,19 @@ id: accordproject-lawyers
...
+As an individual, contributing to the Accord Project would be a great opportunity to **learn** about smart legal contracts. Through the Accord Project, you can understand the foundations of open source technologies and learn how to develop smart agreements.
+
+If your organization wants to become a member of the Accord Project, please join our community [through this link](https://www.accordproject.org/membership).
+
+
+## How to use this documentation?
+
+If you are new to the Accord Project, please read the [Key Concepts](accordproject-concepts) page. This will allow you to understand the three components of a template (text, model, and logic) and how they work together.
+
+If you want to create a template yourself, please see the [Legal-Tech Tutorials - Authoring in Template Studio](tutorial-latedelivery) page for a step-by-step guide on how to create your first template. The tutorial will provide an accessible starting point for those without significant development experience to begin building smart legal contracts using the Accord Project technology.
```
Comment (jeromesimeon) [inline docs/accordproject-laywers.md]: I would point to the video first for lawyers (the `online tour` section)

File: docs/accordproject-developers.md [hunk 7]
```diff
@@ -3,4 +3,22 @@ id: accordproject-developers
...
+
+If you want to dive into our technology stack, you can find more information about:
+- Our software implementations for the Accord Project, [Cicero](https://github.com/accordproject/cicero);
+- The Accord Project markup, [CiceroMark](markup-cicero);
+- Our modeling language used, [Concerto](model-concerto); and
+- Our programming language, [Ergo](logic-ergo).
```
Comment (irmerk) [inline docs/accordproject-developers.md]: I would think this goes better as something like: 
```
If you want to dive into the Accord Project technology stack, you can find more information about:
- Software implementation: [Cicero](https://github.com/accordproject/cicero);
- Markup: [CiceroMark](markup-cicero);
- Modeling language: [Concerto](model-concerto); and
- Programming language: [Ergo](logic-ergo).
```

File: docs/accordproject-developers.md [hunk 8]
```diff
@@ -3,4 +3,22 @@ id: accordproject-developers
...
-**TBD**
+## Why is the Accord Project relevant for developers?
+The Accord Project provides a universal format for smart legal contracts, and this format is embodied in a variety of open source projects that comprise the Accord Project technology stack. Input from developers are crucial for the Accord Project.
+
+Developers can contribute by translating legal text into corresponding computer code, creating Accord Project templates to be used by lawyers and businesses. In addition, developers can provide input on the development of the technology stack components: language, models, templating, and other tools.
+
+If this interests you, please visit our [Technology Working Group](https://www.accordproject.org/working-groups/technology) page, and join our [slack channel](https://accord-project-slack-signup.herokuapp.com/)!
+
+
+## How to use this documentation?
```
Comment (irmerk) [inline docs/accordproject-developers.md]: How to navigate this documentation? Documentation navigation?

File: docs/accordproject-concepts.md [hunk 9]
```diff
@@ -5,31 +5,31 @@ title: Key Concepts
 
 ## What is an Accord Project Template?
 
-An Accord Project template ties legal text (for a clause or contract) to computer code. It is composed of three elements: 
+An Accord Project template ties legal text (for a clause or contract) to computer code. It is composed of three elements:
```
Comment (irmerk) [inline docs/accordproject-concepts.md]: Is "(for a clause or contract)" necessary here?

File: docs/accordproject-concepts.md [hunk 10]
```diff
@@ -5,31 +5,31 @@ title: Key Concepts
...
 
 - **The Template Text**: the natural language of the template
-- **The Template Model**: the data model that backs the template 
+- **The Template Model**: the data model that backs the template
 - **The Template Logic**: the executable business logic for the template
```
Comment (irmerk) [inline docs/accordproject-concepts.md]: These could be links to Cicero, Concerto, and Ergo?

Comment (jeromesimeon) [reply docs/accordproject-concepts.md]: > These could be links to Cicero, Concerto, and Ergo?
> 
> Also, one of the `the`'s could be removed, preferably the first ones to make it
> 
> ```
> Template Text
> Template Model
> Template Logic
> ```

I think the first `The` could be removed.

The pointers to Cicero / Ergo / Concerto is given later in that same page, the idea here is to introduce the _concepts_ not the implementation of those concepts.

File: docs/accordproject-concepts.md (hunk 1 above)
Comment (jeromesimeon) [reply docs/accordproject-concepts.md]: > Hahaha sounds good ;) (makes note to self - nothing with _linear_ next time...)

Well I could suggest to use linear when it's actually _linear_ ! 😉

File: docs/accordproject-concepts.md [hunk 11]
```diff
@@ -5,31 +5,31 @@ title: Key Concepts
...
 
-> Note that we use the computing term 'execution' here, which means run by a computer. This is distinct from the legal term 'execution' which usually refers to the process of signing an agreement.
+> Note that we use the computing term 'execution' here, which means run by a computer. This is distinct from the legal term 'execution', which usually refers to the process of signing an agreement.
 
 ### Cicero
 
-The main software implementation for the Accord Project is called [Cicero](https://github.com/accordproject/cicero). It allows users of the accord project to create and manipulate software templates which tie together all three components in the triangle above (i.e., the text, model and logic).
+The main software implementation for the Accord Project is called [Cicero](https://github.com/accordproject/cicero). It allows users of the Accord Project to create and manipulate software templates which tie together all three components in the triangle above (i.e., the text, model and logic).
```
Comment (irmerk) [inline docs/accordproject-concepts.md]: This may be nit-picky but the second 'Accord Project' feels repetitive.

File: docs/accordproject-concepts.md [hunk 12]
```diff
@@ -5,31 +5,31 @@ title: Key Concepts
...
 
-_More information about how to install Cicero and get started with Accord project templates can be found in the [Installation](accordproject-installation) Section of this documentation._
+_More information about how to install Cicero and get started with Accord Project templates can be found in the [Installation](accordproject-installation) Section of this documentation._
 
 ## Template Text
 
 ![Template Text](assets/020/template_text.png)
 
-The template text is the natural language of the clause or contract. It can include markup to indicate variables (or parameters) for that template.
+The template text is the natural language of the clause or contract. It can include markup to indicate variables (or parameters) for that template. The model reflects the data referred to in the text, and the logic reflects the business logic in the text - this will be explained in the later sections.
```
Comment (irmerk) [inline docs/accordproject-concepts.md]: > Both the model, which reflects the data referred to in the text, and the logic, which reflects the business logic in the text, will be explained in future sections.

File: docs/accordproject-concepts.md [hunk 13]
```diff
@@ -54,9 +54,9 @@ obligations under this agreement, detailed in {{attachment}}, attached
 to this agreement.
 ```
 
-The text is written in plain English with variables between `{{` and `}}` and highlighted in blue. Using variables is useful so the same template can be used in different agreements by simply replacing them by different values.
+The text is written in plain English, with variables between `{{` and `}}` and highlighted in blue. Using variables is useful so the same template can be used in different agreements by simply replacing them with different values.
```
Comment (irmerk) [inline docs/accordproject-concepts.md]: Remove "simply"

File: docs/accordproject-concepts.md [hunk 14]
```diff
@@ -5,31 +5,31 @@ title: Key Concepts
...
+- **The Template Model**: the data model that backs the template
 - **The Template Logic**: the executable business logic for the template
 
 ![Accord Project Template](assets/020/template.png)
 
-When combined these three elements allow templates to be edited, validated, verified, and then executed on any computer platform (on your own machine, on a Cloud platform, on Blockchain, etc).
+The three components (Text - Model - Logic) can also be intuitively understood as a **progression**, from _human-readable_ legal text to _machine-readable_ code. When combined these three elements allow templates to be edited, validated, verified, and then executed on any computer platform (on your own machine, on a Cloud platform, on Blockchain, etc).
 
-> Note that we use the computing term 'execution' here, which means run by a computer. This is distinct from the legal term 'execution' which usually refers to the process of signing an agreement.
+> Note that we use the computing term 'execution' here, which means run by a computer. This is distinct from the legal term 'execution', which usually refers to the process of signing an agreement.
```
Comment (irmerk) [inline docs/accordproject-concepts.md]: > Note the use of the computing term ...
Removing "we"

File: docs/accordproject-concepts.md [hunk 15]
```diff
@@ -83,7 +83,7 @@ to this agreement.
 
 ### CiceroMark
 
-CiceroMark is the markup format in which the text for Accord Project templates is written. It defines notations (such as the `{{` and `}}` notation for variables that we saw in the acceptance of delivery clause above) which allows a computer to make sense of your templates.
+CiceroMark is the markup format in which the text for Accord Project templates is written. It defines notations (such as the `{{` and `}}` notation for variables that we saw in the **Acceptance of Delivery** clause above) which allows a computer to make sense of your templates.
```
Comment (irmerk) [inline docs/accordproject-concepts.md]: > ... notation for variables that were used in the **Acceptance ...

Removing "we"

File: docs/accordproject-concepts.md [hunk 16]
```diff
@@ -93,13 +93,32 @@ _More information about the Accord Project markup can be found in the [CiceroMar
...
-Thanks to that model, the computer now knows that the `shipper` variable (`"Party A"` in the example above) contains a reference to an `Organization`, that variable `receiver` (`"Party B"` in the example above) also contains a reference to an Organization, that variable `deliverable` (`"Widgets"` in the example above) can contain any text description, etc.
+```ergo
+/* The template model */
+asset AcceptanceOfDeliveryClause extends AccordClause {
+  /** the shipper of the goods*/
+  --> Organization shipper
+  /** the receiver of the goods */
+  --> Organization receiver
+
+  /** what are we delivering? */
```
Comment (irmerk) [inline docs/accordproject-concepts.md]: > what is being delivered?

Removing "we"

File: docs/accordproject-concepts.md [hunk 17]
```diff
@@ -93,13 +93,32 @@ _More information about the Accord Project markup can be found in the [CiceroMar
 
 ![Template Model](assets/020/template_model.png)
 
-Compared to a standard document template (e.g., in Word or pdf), Accord Project templates associate a _model_ to the natural language text. The model lets you categorize variables (is it a number, a monetary amount, a date, a reference to a business or organization, etc.). This model is crucial in that it allows the computer to make sense of the information contained in your template.
+Compared to a standard document template (e.g., in Word or pdf), Accord Project templates associate a _model_ to the natural language text. The model acts as a bridge between the text and logic; it gives the users an overview of the components, as well as the traits of different components.
```
Comment (irmerk) [inline docs/accordproject-concepts.md]: Unsure what is meant by 'Compated to...', is this in contrast? Similarity?

File: docs/accordproject-concepts.md [hunk 18]
```diff
@@ -93,13 +93,32 @@ _More information about the Accord Project markup can be found in the [CiceroMar
...
 
-The following screenshot shows the model for the acceptance of delivery clause above.
+The model lets you categorize variables (is it a number, a monetary amount, a date, a reference to a business or organization, etc.). This model is crucial, because it allows the computer to make sense of the information contained in your template.
 
-![Example Model](/img/model-vscode.png)
+The following shows the model for the **Acceptance of Delivery** clause above.
```
Comment (irmerk) [inline docs/accordproject-concepts.md]: I would avoid directional references such as "above" or "below". Could maybe use "previously"?

File: docs/accordproject-concepts.md [hunk 19]
```diff
@@ -93,13 +93,32 @@ _More information about the Accord Project markup can be found in the [CiceroMar
...
 
-Thanks to that model, the computer now knows that the `shipper` variable (`"Party A"` in the example above) contains a reference to an `Organization`, that variable `receiver` (`"Party B"` in the example above) also contains a reference to an Organization, that variable `deliverable` (`"Widgets"` in the example above) can contain any text description, etc.
+```ergo
+/* The template model */
+asset AcceptanceOfDeliveryClause extends AccordClause {
+  /** the shipper of the goods*/
+  --> Organization shipper
+  /** the receiver of the goods */
+  --> Organization receiver
```
Comment (irmerk) [inline docs/accordproject-concepts.md]: Formatting could be synced up with the rest of this code snippet by having a space in between each of these.

File: docs/accordproject-concepts.md [hunk 20]
```diff
@@ -93,13 +93,32 @@ _More information about the Accord Project markup can be found in the [CiceroMar
...
+
+  /* how long does the receiver have to inspect the goods? */
+  o Integer businessDays
+
+  /** additional information */
+  o String attachment
+}
+```
+
+Thanks to the model, the computer now knows that the `shipper` variable (`"Party A"` in the example above) and the `receiver` variable (`"Party B"` in the example above) are both `Organization` types; that variable `businessDays` (`10` in the example above) is an `Integer` type; and that variable `deliverable` (`"Widgets"` in the example above) is a `String` type, and can contain any text description.
```
Comment (irmerk) [inline docs/accordproject-concepts.md]: Avoid "above"

Also could make a reference to the Concerto section coming after, because this feels like it may be separate from the next section, and I may be left confused at what an "Organization type" is, etc.

File: docs/accordproject-concepts.md [hunk 21]
```diff
@@ -113,19 +132,41 @@ _More information about Concerto can be found in the [Concerto Modeling](model-c
 
 While the combination of text and model already sets Accord Project template apart from a more traditional document format, the logic is what allows templates to really _come alive_. The logic allows a developer to associate _behavior_ to the template (specify what happens when a delivery is received late, check conditions for payment, automate interest rate calculations, determine if there has been a breach of contract, etc).
 
-The following screenshot shows the logic for the acceptance of delivery clause above.
-
-![Example Logic](/img/ergo-vscode.png)
+The following shows the logic for the **Acceptance of Delivery** clause above.
```
Comment (irmerk) [inline docs/accordproject-concepts.md]: Avoid "above"

Also, the paragraph before this should be "... and model already sets Accord Project templates apart from a more traditional ..."

File: docs/accordproject-concepts.md [hunk 22]
```diff
@@ -113,19 +132,41 @@ _More information about Concerto can be found in the [Concerto Modeling](model-c
...
+      status : status,
+      shipper : contract.shipper,
+      receiver : contract.receiver
+    }
+  }
+}
+```
 
-That logic specify what conditions should be met for a delivery to be accepted (checking whether the delivery is timely or not, has a mandatory inspection for that delivery occurred or not, etc).
+The above logic specifies what conditions should be met for a delivery to be accepted. The code above checks whether delivery has already been made; whether the acceptance is timely, within the specified inspection date; and whether the inspection has passed or not.
```
Comment (irmerk) [inline docs/accordproject-concepts.md]: Avoid "above"

File: docs/accordproject-laywers.md [hunk 23]
```diff
@@ -3,4 +3,19 @@ id: accordproject-lawyers
...
+Even the best code would be useless if it does not meet industry needs and requirements, so input from the legal community to steer the direction of smart legal contract technology is crucial. By creating templates for contract clauses or reviewing templates, lawyers can **influence** the direction of the Accord Project. In addition, feedback using domain-specific expertise can enhance the value and usability of the Accord Project templates to users.
+
+If this interests you, please visit our [Lifecycle and Industry Working Groups](https://www.accordproject.org/liwg) page for more information.
+
+As an individual, contributing to the Accord Project would be a great opportunity to **learn** about smart legal contracts. Through the Accord Project, you can understand the foundations of open source technologies and learn how to develop smart agreements.
+
+If your organization wants to become a member of the Accord Project, please join our community [through this link](https://www.accordproject.org/membership).
+
+
+## How to use this documentation?
```
Comment (irmerk) [inline docs/accordproject-laywers.md]: How to navigate this documentation? Documentation navigation?

File: docs/accordproject-concepts.md (hunk 16 above)
Comment (jeromesimeon) [reply docs/accordproject-concepts.md]: @irmerk @4katie As much as possible I would try to keep those examples be the exactly what's in the corresponding template. If we make changes here, then please make the same changes to the template in the template library as well.

**Comment by irmerk** [comment]
Added a bunch of comments, this is looking so good @4katie!

File: docs/accordproject-developers.md (hunk 7 above)
Comment (jeromesimeon) [reply docs/accordproject-developers.md]: Please use the same titles as in the corresponding Sections.

File: docs/accordproject-concepts.md (hunk 16 above)
Comment (irmerk) [reply docs/accordproject-concepts.md]: Ah didn't think to check that. Then skip this comment. I still think "we" should be avoided in the rest of the docs if possible.

File: docs/accordproject-developers.md (hunk 7 above)
Comment (irmerk) [reply docs/accordproject-developers.md]: Sure. The current change has the bullets seeming awkward and out of sync.

**Comment by 4katie** [comment]
Hi @irmerk and @jeromesimeon, thanks for such quick & helpful feedback!

I think I made all of the changes requested except for:
- the paragraph introducing Cicero (not sure if this is sufficient or satisfactory) and
- making the link between Template Model and Concerto. 

Any comments / feedback / suggestions would be super helpful! :D
